src/components/List.jsx

```diff
@@ -27,32 +28,54 @@ export const List = ({ token }) => {
     });
   };
 
-  const handleClick = (data, e) => {
-    updateDocument(data.id);
+  const handleClick = (doc) => {
+    updateDocument(doc.id);
+  };
+
+  const handleFilterChange = (e) => {
+    setFilterText(e.target.value);
+  };
+
+  const clearFilter = () => {
+    setFilterText('');
   };
-  
   return (
     <div className="welcoming">
       <h1>Smart Shopping List</h1>
       <strong> Shareable List Token : {token} </strong>
+
+      <div style={{ marginTop: '1em' }}>
+        <input
+          placeholder="Start typing here..."
+          value={filterText}
+          onChange={handleFilterChange}
+        />
+        <button style={{ marginLeft: '10px' }} onClick={() => clearFilter()}>
+          X
+        </button>
```

Comment:
Not a must change. For future reference, you can call the setFilterText function in the onClick
```suggestion
        <button style={{ marginLeft: '10px' }} onClick={() => setFilterText('')}>
          X
        </button>
```